addons/product_expiry/models/stock_move_line.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import datetime
+
+from odoo import fields, models
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    def _default_expiration_date(self):
+        if self.env.context.get('active_model') != 'stock.move':
+            return
+        move = self.env['stock.move'].browse(self.env.context.get('active_id'))
+        if not move.exists():
+            return
+        product = move.product_id
+        if not product.use_expiration_date:
+            return
+        return fields.Datetime.today() + datetime.timedelta(days=product.expiration_time)
```

Comment:
default method should not be specific to a record, maybe this should be converted to an onchange on product_id?